### PR TITLE
Make JAuthentication autoloaded, split 2 classes to 2 files

### PR DESCRIPTION
**Pull request by mbabker**
#### Summary of Changes

The `JAuthentication` class requires manual import and the file that holds this class contains two classes (`JAuthentication` and `JAuthenticationResponse`).  Move the file so it can be autoloaded and split the response class to a new file to follow the one class per file convention.
#### Testing Instructions

User authentication still works correctly, on update the old file is deleted.

### CODE REVIEW

File: administrator/components/com_admin/script.php [hunk 1]
```diff
@@ -1424,6 +1424,8 @@ public function deleteUnexistingFiles()
 			'/administrator/templates/isis/js/jquery.js',
 			'/administrator/templates/isis/js/bootstrap.min.js',
 			'/media/system/js/permissions.min.js',
+			// Joomla __DEPLOY_VERSION__
```
Comment (andrepereiradasilva) [inline administrator/components/com_admin/script.php]: this (https://github.com/joomla/joomla-cms/pull/10883) needs to be merged for `__DEPLOY_VERSION__` to work

Comment (mbabker) [inline administrator/components/com_admin/script.php]: I'm counting on that being merged before this PR ever makes it to a stable release 😉

Comment (andrepereiradasilva) [inline administrator/components/com_admin/script.php]: :wink:

**Comment by hardiktailored** [comment]
I have tested this item :white_check_mark: successfully on 3233daa8bef5befaada34bc09e75863eec12eb5d<br /><br />I have Joomla Authentication Plugin enabled and user authentication works correctly after applying patch too.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11023">issues.joomla.org/joomla-cms/11023</a>.</sub>

**Comment by rahultomar** [comment]
I have not tested this item.<br /><br />Tested , and working fine<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11023">issues.joomla.org/joomla-cms/11023</a>.</sub>

**Comment by rahultomar** [comment]
I have tested this item :white_check_mark: successfully on 3233daa8bef5befaada34bc09e75863eec12eb5d<br /><br />Tested , and working fine<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11023">issues.joomla.org/joomla-cms/11023</a>.</sub>

**Comment by gunjanpatel** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11023">issues.joomla.org/joomla-cms/11023</a>.</sub>

**Comment by wilsonge** [comment]
Merged with 761bf53 - thanks!

**Pull request closed without merging**